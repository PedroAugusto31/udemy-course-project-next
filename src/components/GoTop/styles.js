import styled, { css } from "styled-components";

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: ${theme.colors.primary};
    color: ${theme.colors.white};
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
    opacity: 0.8;
  `}
`;
