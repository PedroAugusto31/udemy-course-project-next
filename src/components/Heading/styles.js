import styled, { css } from "styled-components";

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.xhuge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.medium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`;

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? "uppercase" : "none"};
`;

export const Title = styled.h1`
  ${({ theme, isLight, size, uppercase }) => css`
    color: ${isLight ? theme.colors.white : theme.colors.primary};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
