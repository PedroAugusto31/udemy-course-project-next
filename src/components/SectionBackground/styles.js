import styled, { css } from "styled-components";

const containerBackgroundActivate = (theme) => css`
  background: ${theme.colors.primary};
  color: ${theme.colors.white};
`;

export const Container = styled.div`
  ${({ theme, background }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primary};
    ${background && containerBackgroundActivate(theme)};

    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
