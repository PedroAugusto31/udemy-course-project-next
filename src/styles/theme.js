export const theme = {
  colors: {
    mainColor: "#5290E3",
    primary: "#0A1128",
    secondary: "#dc143c",
    white: "#FFFFFF",
    mediumGray: "#DDDDDD",
  },
  font: {
    family: {
      default: "'Open Sans', sans-serif",
      secondary: "'Montserrat', sans-serif",
    },
    sizes: {
      xsmall: "0.8rem",
      small: "1.6rem",
      medium: "2.4rem",
      large: "3.2rem",
      xlarge: "4.0rem",
      xxlarge: "4.8rem",
      huge: "5.6rem",
      xhuge: "6.4rem",
    },
  },
  media: {
    small: "(max-width: 526px)",
    medium: "(max-width: 768px)",
  },
  spacings: {
    xsmall: "0.8rem",
    small: "1.6rem",
    medium: "2.4rem",
    large: "3.2rem",
    xlarge: "4.0rem",
    xxlarge: "4.8rem",
    huge: "5.6rem",
    xhuge: "6.4rem",
  },
};
