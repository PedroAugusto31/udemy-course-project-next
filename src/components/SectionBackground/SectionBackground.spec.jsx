import { screen } from "@testing-library/react";
import { SectionBackground } from ".";
import { renderTheme } from "../../styles/render-theme";

describe("<SectionBackground />", () => {
  it("Should render with correct background dark", () => {
    const { container } = renderTheme(
      <SectionBackground background>Children</SectionBackground>,
    );
    expect(screen.getByText("Children")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it("Should render with background light", () => {
    const { container } = renderTheme(
      <SectionBackground>Children</SectionBackground>,
    );
    expect(container.firstChild).toBeInTheDocument();
  });
});
