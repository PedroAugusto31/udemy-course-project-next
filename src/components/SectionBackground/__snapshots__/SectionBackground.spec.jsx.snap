// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionBackground /> Should render with correct background dark 1`] = `
.c0 {
  background: #FFFFFF;
  color: #FFFFFF;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 1.6rem;
  width: 100%;
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      Children
    </div>
  </div>
</div>
`;
